lib/compat/wordpress-6.3/html-api/class-gutenberg-html-tag-processor-6-3.php

```diff
@@ -1442,20 +1443,22 @@ private function apply_attributes_updates() {
 			$tail_delta = 0;
 
 			foreach ( $this->lexical_updates as $diff ) {
-				$update_head = $bookmark->start >= $diff->start;
-				$update_tail = $bookmark->end >= $diff->start;
-
-				if ( ! $update_head && ! $update_tail ) {
+				if ( ! $bookmark->start >= $diff->start && ! $bookmark->end >= $diff->start ) {
```

Comment:
we're now dangerously close to some double-negativo here. would the intent of this be clearer?

```php
if ( $bookmark->start < $diff->start && $bookmark->end < $diff->start ) {
	break;
}
```

Reply:
Eh, don't mind me, I'm just copy-pasting [your code](https://github.com/WordPress/gutenberg/pull/47559#discussion_r1103140849) 😆 

But yeah, it did occur to me that it might not be the least unclear thing to do 😏 Happy to change it!

Reply:
I think in some ways we're stating that as a constraint, so it should be enough, but I thought it best to leave that safety in as a fallback, since a bug here implies breaking HTML syntax.

also I considered flipping the conditions to check the end first, but then that also ready confusingly, which is what led me to stick with the check I wrote above, a correction apparently to my earlier tangle of code I suggested 😆

Reply:
Changed in d91a17fe3e1f2d1426ccf32ea46646e0e839601d.